src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -4262,12 +4282,18 @@ private bool HasTopLevelNullabilityConversion(TypeWithAnnotations source, TypeWi
                     // is converting to the same type including nullability, skip the conversion
                     // to avoid reporting redundant warnings. Also check useLegacyWarnings
                     // since that value was used when reporting warnings for the explicit cast.
-                    if (useLegacyWarnings && operandOpt?.Kind == BoundKind.Conversion)
+                    // Don't skip the node when it's a user-defined conversion, as identity conversions
+                    // on top of user-defined conversions means that we're coming in from VisitUserDefinedConversion
+                    // and that any warnings caught by this pass of VisitConversion won't be redundant.
```

Comment:
> this pass of VisitConversion [](start = 54, length = 29)

I didn't understand this. We're doing multiple passes of VisitConversion?

Reply:
Yes. See `VisitUserDefinedConversion`, which will call `VisitConversion` multiple times.

---
In reply to: [282235669](https://github.com/dotnet/roslyn/pull/35544#discussion_r282235669) [](ancestors = 282235669)

Reply:
I'll change `pass` to `recursive call` to be clearer.

---
In reply to: [282267832](https://github.com/dotnet/roslyn/pull/35544#discussion_r282267832) [](ancestors = 282267832,282235669)